### PR TITLE
add favicon to documentation assets and make.jl

### PR DESCRIPTION
**Pull request by wolthom**
This commit fixes issue #2477

First PR, so if I didn't follow any standard procedure, all feedback is welcome.

### CODE REVIEW

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
All looks good! Thank you for contributing. (let us just wait for CI to finish)

**Comment by wolthom** [comment]
Looks like the CI passed now. Do I need to do anything for the branch to be merged?

File: docs/make.jl [hunk 1]
```diff
@@ -14,7 +14,10 @@ makedocs(
     doctest = false,
     clean = false,
     sitename = "DataFrames.jl",
-    format = Documenter.HTML(canonical = "https://juliadata.github.io/DataFrames.jl/stable/"),
+    format=Documenter.HTML(
+        canonical="https://juliadata.github.io/DataFrames.jl/stable/",
+        assets=["assets/favicon.ico"]
+    ),
```
Comment (nalimilan) [inline docs/make.jl:20]:
```suggestion
    format = Documenter.HTML(
        canonical = "https://juliadata.github.io/DataFrames.jl/stable/",
        assets = ["assets/favicon.ico"]
    ),
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**